app/models/sample.rb

```diff
@@ -267,6 +273,27 @@ def check_host_genome
     s3_preload_result_path.strip!
   end
 
+  def concatenate_input_parts
```

Comment:
you are doing this for every sample.save?

Reply:
thanks, added a condition